verification/verify.go

```diff
@@ -103,13 +110,29 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		return nil, nil
 	}
 
-	// TODO sig checking; extract from broadcast.go
+	if params.Results != nil {
+		segHashes := make([][]byte, len(params.Results.Segments))
+		if len(segHashes) == len(params.Renditions) {
```

Comment:
Do we need this if statement if we instead just iterate through `params.Renditions` instead of `params.Results.Segments`? Then, if `len(params.Renditions) != len(params.Results.Segments)` the comparison of `segHashes` and `params.Results.Segments` would fail below.

Reply:
We still need the `if` statement because after the loop we need to be able to read `params.Results.Sig` without a `nil` pointer dereference

Reply:
Apologies, I was looking at the wrong `if` statement at first.  I do still believe that we need it, however.  When the following diff is applied:
```diff
diff --git a/verification/verify.go b/verification/verify.go
index 20c3f7ee..a0701cac 100644
--- a/verification/verify.go
+++ b/verification/verify.go
@@ -112,11 +112,9 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 
        if params.Results != nil {
                segHashes := make([][]byte, len(params.Results.Segments))
-               if len(segHashes) == len(params.Renditions) {
-                       for i := range params.Results.Segments {
+               for i := range params.Renditions {
                        segHashes[i] = crypto.Keccak256(params.Renditions[i])
                }
-               }
 
                // Might not have seg hashes if results are directly uploaded to the broadcaster's OS
                if len(segHashes) != len(params.Results.Segments) &&
```
we get the following `nil` pointer dereference in the line `segHashes[i] = crypto.Keccak256(params.Renditions[i])`, line 116:
```
--- FAIL: TestVerifier_Verify (0.00s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 788 [running]:
testing.tRunner.func1(0xc0006c6600)
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:874 +0x3a3
panic(0x19989c0, 0xc0005ae300)
	/home/mk/.gvm/gos/go1.13.7/src/runtime/panic.go:679 +0x1b2
github.com/livepeer/go-livepeer/verification.(*SegmentVerifier).Verify(0xc000076e80, 0xc00021a4d0, 0x1, 0xc0003be140, 0x1834100)
	/home/mk/git/livepeer/go-livepeer/verification/verify.go:116 +0x8ec
github.com/livepeer/go-livepeer/server.verify(0xc000076e80, 0xc00068fef8, 0xc0004b7180, 0xc0004bc2c0, 0xc0000d2320, 0xc0006a7560, 0x1, 0x1, 0x1ce1601, 0xc0006a77b0)
	/home/mk/git/livepeer/go-livepeer/server/broadcast.go:533 +0x1ea
github.com/livepeer/go-livepeer/server.TestVerifier_Verify(0xc0006c6600)
	/home/mk/git/livepeer/go-livepeer/server/broadcast_test.go:809 +0x7aa
testing.tRunner(0xc0006c6600, 0x1ab9cc0)
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:960 +0x350
exit status 2
FAIL	github.com/livepeer/go-livepeer/server	3.115s
```

Reply:
I think the index out of range issue would be resolved if `segHashes` was initialized with a length of `len(params.Renditions)` rather than `len(params.Results.Segments)`.

Reply:
doing so results in another range error, then i changed it to range over `params.Renditions` instead and that causes a nil pointer dereference.  Is there a reason why we can't just keep the check that `params.Results != nil` here?  We'll blow up on a nil pointer then we try to read `params.Results.Sig`

Reply:
The nil pointer deference is caused by the call to `lpcrypto.VerifySig()` which references `params.Orchestrator.TicketParams.Recipient` and `params.Orchestrator` is nil in the test that triggers the nil pointer deference. As mentioned in [the pixel count PR](https://github.com/livepeer/go-livepeer/pull/1360/files#r378488954), we'll need an on-chain check before running sig verification now - adding this check should also resolve the nil pointer deference in the tests i.e. `params.Orchestrator != nil && params.Orchestrator.TicketParams != nil `. Ideally we wouldn't need that `params.Orchestrator != nil` check since it would be non-nil in practice, I think a few of the tests have it set to nil so including the check may be easier.

Reply:
I believe this may be resolved by db98e762579c107cf87067e27c20c3e10bd68bcc and 0e7f71b18090d03153ba8e063db72595d3a0e8b1